lib/DDG/Spice/Holiday.pm

```diff
@@ -2,68 +2,87 @@ package DDG::Spice::Holiday;
 
 # ABSTRACT: Query timeanddate.com for a holiday
 
+use strict;
 use DDG::Spice;
-use POSIX qw(strftime);
+use Locale::Country;
 
-triggers start => 'when is', 'when was', 'what day is', 'what day was';
+spice is_cached => 1;
+spice proxy_cache_valid => "200 30d";
+spice wrap_jsonp_callback => 0;
 
 spice from => '([^/]+)/([^/]+)/([^/]+)/([^/]+)';
-spice to => 'http://www.timeanddate.com/scripts/ddg.php?m=whenis&c=$2&q=$3&y=$4&callback={{callback}}';
+spice to => 'http://www.timeanddate.com/scripts/ddg.php?m=whenis&c=$1&q=$2&y=$3&callback={{callback}}';
 
-handle query_lc => sub {
-    return unless ($_);
+my @triggers  = ('when is', 'what day is');
+my $countries = join('|', share("countries.txt")->slurp(chomp => 1));
 
-    my ($t, $q, $c, $y);
+triggers start => @triggers;
```

Comment:
@sekhavati now that we have a list of all supported holidays, the triggers should be the unique set of holiday names. That way "father's day" or "father's day 2017" will trigger this and show the correct information.

Based on the user's country we can check if the named holiday exists for their country and if not, fallback to the US for the API call.

Does this make sense?

Can you create a "holidays.txt" file with the unique names of all holidays across all countries? We should probably build it with a script that uses the existing `.hols` files. Also, it may be best to rename all those to a standard file extension? I think we can even put them in their own subdirectory

e.g. `/share/spice/holiday/countries/al.txt`